tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/impl/BlobStoreBackedReadHandleImpl.java

```diff
@@ -105,6 +118,11 @@ public LedgerMetadata getLedgerMetadata() {
                 long nextExpectedId = firstEntry;
                 try {
                     while (entriesToRead > 0) {
+                        State state = STATE_UPDATER.get(this);
+                        if (state == State.Closed) {
+                            log.warn("Reading a closed read handler. Ledger ID: {}, Read range: {}-{}", ledgerId, firstEntry, lastEntry);
+                            throw new BKException.BKUnexpectedConditionException();
```

Comment:
Should this complete the `promise` exceptionally?

Reply:
The `promise` is completed in the `catch` section.